src/manager.rs

```diff
@@ -28,90 +29,89 @@ pub struct PartitionedService {
-        }
-    }
+    pub fn start(config: Arc<Config>, partition: u16) -> (Self, JoinHandle<()>) {
+        let config_store = Arc::new(ConfigStore::new_rw());
 
-    pub fn get_config_store(&self) -> Arc<RwConfigStore> {
-        self.config_store.clone()
-    }
-
-    /// Begin scheduling checks for this partition.
-    pub fn start(&self) -> CancellationToken {
         let checker = Arc::new(HttpChecker::new());
-
         let producer = Arc::new(KafkaResultsProducer::new(
-            &self.config.results_kafka_topic,
-            KafkaConfig::new_config(self.config.results_kafka_cluster.to_owned(), None),
+            &config.results_kafka_topic,
+            KafkaConfig::new_config(config.results_kafka_cluster.to_owned(), None),
         ));
 
-        let config_store = self.get_config_store();
-        let partition = self.partition;
+        let waiter_config_store = config_store.clone();
 
         // TODO(epurkhiser): We may want to wait to start the scheduler until "booting" completes,
         // otherwise we may execute checks for old configs in a partition that are removed later in
         // the log.
-        tokio::spawn(async move { wait_for_partition_boot(config_store, partition).await });
+        tokio::spawn(async move { wait_for_partition_boot(waiter_config_store, partition).await });
 
-        run_scheduler(
-            self.partition,
-            self.get_config_store(),
+        let shutdown_signal = CancellationToken::new();
+        let scheduler_join_handle = run_scheduler(
+            partition,
+            config_store.clone(),
             checker,
             producer,
-            self.shutdown_signal.clone(),
+            shutdown_signal.clone(),
         );
-        self.shutdown_signal.clone()
+
+        let service = Self {
+            config,
+            partition,
+            config_store,
+            shutdown_signal,
+        };
+
+        (service, scheduler_join_handle)
     }
 
     pub fn get_partition(&self) -> u16 {
         self.partition
     }
 
+    pub fn get_config_store(&self) -> Arc<RwConfigStore> {
+        self.config_store.clone()
+    }
+
     pub fn stop(&self) {
-        self.shutdown_signal.cancel()
+        self.shutdown_signal.cancel();
     }
 }
 
+#[derive(Debug)]
+struct ServiceHandle {
+    service: Arc<PartitionedService>,
+    join_handle: JoinHandle<()>,
+}
+
 #[derive(Debug)]
 pub struct Manager {
     config: Arc<Config>,
-    services: RwLock<HashMap<u16, Arc<PartitionedService>>>,
+    services: RwLock<HashMap<u16, ServiceHandle>>,
     shutdown_signal: CancellationToken,
 }
 
 impl Manager {
-    pub fn new(config: Arc<Config>) -> Self {
-        Self {
-            config,
-            services: RwLock::new(HashMap::new()),
-            shutdown_signal: CancellationToken::new(),
-        }
-    }
-
-    /// Instantiates a new manager with a single partition and a default config
-    pub fn new_simple() -> Self {
```

Comment:
Removed this helper. Felt weird to have a test helper like this in the impl anyway.